From 6f764cceeb7350d8da33765da07462578f07e988 Mon Sep 17 00:00:00 2001
From: Heming Zhao <heming.zhao@suse.com>
Date: Fri, 8 Oct 2021 09:33:36 +0800
Subject: [PATCH 08/14] man: add reload_config in dlm_tool & dlm.conf

reload_config is a new command of dlm_tool. We could use this command
to change dlm_controld some settings on the fly.

This patch modifies man page to add usage for this new command.

Signed-off-by: Heming Zhao <heming.zhao@suse.com>
---
 dlm_controld/dlm.conf.5 | 42 ++++++++++++++++++++++++++++++-----------
 dlm_tool/dlm_tool.8     | 10 ++++++++++
 2 files changed, 41 insertions(+), 11 deletions(-)

diff --git a/dlm_controld/dlm.conf.5 b/dlm_controld/dlm.conf.5
index 1ce0c6445ee1..99e302a92bd2 100644
--- a/dlm_controld/dlm.conf.5
+++ b/dlm_controld/dlm.conf.5
@@ -34,7 +34,9 @@ protocol=tcp
 
 Options:
 
-log_debug
+daemon_debug(*)
+.br
+log_debug(*)
 .br
 protocol
 .br
@@ -42,23 +44,23 @@ bind_all
 .br
 mark
 .br
-debug_logfile
+debug_logfile(*)
 .br
 enable_plock
 .br
-plock_debug
+plock_debug(*)
 .br
-plock_rate_limit
+plock_rate_limit(*)
 .br
 plock_ownership
 .br
-drop_resources_time
+drop_resources_time(*)
 .br
-drop_resources_count
+drop_resources_count(*)
 .br
-drop_resources_age
+drop_resources_age(*)
 .br
-post_join_delay
+post_join_delay(*)
 .br
 enable_fencing
 .br
@@ -66,15 +68,33 @@ enable_concurrent_fencing
 .br
 enable_startup_fencing
 .br
-enable_quorum_fencing
+enable_quorum_fencing(*)
 .br
-enable_quorum_lockspace
+enable_quorum_lockspace(*)
 .br
-repeat_failed_fencing
+repeat_failed_fencing(*)
 .br
 enable_helper
 .br
 
+Option with tail (*) means this item can be reload on the fly by "dlm_tool reload_config". See \fB"Reload config"\fP for more examples.
+
+.SH Reload config
+
+dlm.conf can be changed then reloaded some settings on the fly, it gives users more flexible and powerful ability to manage dlm.
+
+The supported reload operations:
+.br
+- add a new item
+.br
+- remove an exist item
+.br
+- commented out an item
+.br
+- change an item value
+
+After modifying settings in dlm.conf, run "dlm_tool reload_config" to take effect. User can use "dlm_tool dump_config" to check the result. Again, only reloadable item can be changed.
+
 .SH Fencing
 
 A fence device definition begins with a
diff --git a/dlm_tool/dlm_tool.8 b/dlm_tool/dlm_tool.8
index 7e82f8908ed6..6c1d471abe68 100644
--- a/dlm_tool/dlm_tool.8
+++ b/dlm_tool/dlm_tool.8
@@ -28,6 +28,10 @@ dlm_tool \- a utility for the dlm and dlm_controld daemon
 .br
 	Dump dlm_controld config settings.
 
+.B reload_config
+.br
+	Reload dlm_controld config settings from dlm.conf.
+
 .BI fence_ack " nodeid"
 .br
 	Quit trying to fence a node.
@@ -134,6 +138,12 @@ Print program version information, then exit
 
 -
 
+.B dlm_tool reload_config
+
+* see examples in \fBdlm.conf(5)\fP
+
+-
+
 .B dlm_tool fence_ack \fInodeid\fP
 
 * run "dlm_tool status -v" to get \fInodeid\fP
-- 
2.33.0

